craft_application/commands/lifecycle.py

```diff
@@ -94,7 +100,16 @@ class LifecycleStepCommand(LifecyclePartsCommand):
 
     @override
     def run_managed(self, parsed_args: argparse.Namespace) -> bool:
-        return not parsed_args.destructive_mode
+        """Return whether the command should run in managed mode or not.
```

Comment:
If you move the contents of `_run_managed` into `LifecyclePartsCommand` you can remove it from `LifecycleStepCommand` altogether and refer to `super().run_managed` in `CleanCommand`